### PR TITLE
Fix setting options when using two way binding (T807456)

### PR DESCRIPTION
**Pull request by dxvladislavvolkov**

### CODE REVIEW

File: src/core/component.test.ts [hunk 1]
```diff
@@ -744,6 +744,28 @@ describe("nested option", () => {
         });
     });
 
+    it("component shouldn't set option if option already set (v-model)", (done) => {
+        new Vue({
+            template:
+                `<test-component v-model="value">` +
```
Comment (IlyaKhD) [inline src/core/component.test.ts]: Is there any difference in comparison with  `<test-component prop1="value">`  ?

File: src/core/configuration.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ interface ExpectedChild {
 
 class Configuration {
 
+    public changedOptions: any;
```
Comment (IlyaKhD) [inline src/core/configuration.ts]: I think it's not the best place for the `changedOptions`. (See comments below)

What is more.. how about renaming it to smth like `innerChanges` or `internalChanges` or `mutedChanges`

File: src/core/configuration.ts [hunk 3]
```diff
@@ -208,7 +210,12 @@ function bindOptionWatchers(config: Configuration, vueInstance: Pick<Vue, "$watc
     const targets = config.getOptionsToWatch();
     if (targets) {
         targets.forEach((optionName: string) => {
-            vueInstance.$watch(optionName, (value) => config.updateValue(optionName, value));
+            vueInstance.$watch(optionName, (value) => {
+                if (config.changedOptions[optionName] !== value) {
```
Comment (IlyaKhD) [inline src/core/configuration.ts]: Better make the `changedOptions` an argument shared between `bindOptionWatchers` and `subscribeOnUpdates`

File: src/core/component.test.ts (hunk 1 above)
Comment (dxvladislavvolkov) [reply src/core/component.test.ts]: it is reproduced only with two-way binding

File: src/core/configuration.ts [hunk 4]
```diff
@@ -204,16 +204,24 @@ class Configuration {
     }
 }
 
-function bindOptionWatchers(config: Configuration, vueInstance: Pick<Vue, "$watch">): void {
+function bindOptionWatchers(config: Configuration, vueInstance: Pick<Vue, "$watch">, innerChanges: any): void {
```
Comment (IlyaKhD) [inline src/core/configuration.ts]:
```suggestion
function bindOptionWatchers(config: Configuration, vueInstance: Pick<Vue, "$watch">, innerChanges: Record<string, any>): void {
```

File: src/core/configuration.ts [hunk 5]
```diff
@@ -204,16 +204,24 @@ class Configuration {
...
+            });
         });
     }
 }
 
-function subscribeOnUpdates(config: Configuration, vueInstance: Pick<Vue, "$emit" | "$props">): void {
+function subscribeOnUpdates(
+    config: Configuration,
+    vueInstance: Pick<Vue, "$emit" | "$props">,
+    innerChanges: any): void {
```
Comment (IlyaKhD) [inline src/core/configuration.ts]:
```suggestion
    innerChanges: Record<string, any>): void {
```

File: src/core/component.test.ts (hunk 1 above)
Comment (IlyaKhD) [reply src/core/component.test.ts]: Ok

**Pull request merged**